### PR TITLE
DOCS-3625: Add section to config topic: parameters controlled by Kafka Streams

### PR DESCRIPTION
**Pull request by JimGalasyn**
Add a section on the parameters that Kafka Streams assigns. Also, a couple of other tweaks, like moving the `rocksdb.config.setter` section to be alphabetically ordered.

This content is ported from https://github.com/confluentinc/docs/pull/3604.

### CODE REVIEW

File: docs/streams/architecture.html [hunk 1]
```diff
@@ -80,6 +80,10 @@ <h3><a id="streams_architecture_tasks" href="#streams_architecture_tasks">Stream
         tasks will be automatically restarted on other instances and continue to consume from the same stream partitions.
     </p>
 
+    <p><b>NOTE:</b> Topic partitions are balanced but may be unevenly distributed across consumers. Kafka Streams uses the
```
Comment (mjsax) [inline docs/streams/architecture.html]: Not sure what `are balanced but may be unevenly distributed across consumers` means?

File: docs/streams/developer-guide/config-streams.html [hunk 2]
```diff
@@ -269,6 +269,11 @@ <h4><a class="toc-backref" href="#id5">bootstrap.servers</a><a class="headerlink
               <td colspan="2">The amount of time in milliseconds, before a request is retried. This applies if the <code class="docutils literal"><span class="pre">retries</span></code> parameter is configured to be greater than 0. </td>
               <td>100</td>
           </tr>
+          <tr class="row-even"><td>rocksdb.config.setter</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:272]: Just a question: what is the `row-even` vs `row-odd` tag? Seems we don't use is consistently (there are multiple `row-even` tags in a row above) and it does also not seem to make any difference how the rows are rendered? Why do we even have this tag?

File: docs/streams/developer-guide/config-streams.html [hunk 3]
```diff
@@ -474,6 +479,57 @@ <h4><a class="toc-backref" href="#id11">num.stream.threads</a><a class="headerli
               </dl>
             </div></blockquote>
         </div>
+        <div class="section" id="rocksdb-config-setter">
+          <span id="streams-developer-guide-rocksdb-config"></span><h4><a class="toc-backref" href="#id20">rocksdb.config.setter</a><a class="headerlink" href="#rocksdb-config-setter" title="Permalink to this headline"></a></h4>
+          <blockquote>
+            <div><p>The RocksDB configuration. Kafka Streams uses RocksDB as the default storage engine for persistent stores. To change the default
+              configuration for RocksDB, implement <code class="docutils literal"><span class="pre">RocksDBConfigSetter</span></code> and provide your custom class via <a class="reference external" href="/{{version}}/javadoc/org/apache/kafka/streams/state/RocksDBConfigSetter.html">rocksdb.config.setter</a>.</p>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: `implement` -> `you can implement` ?

File: docs/streams/developer-guide/config-streams.html [hunk 4]
```diff
@@ -474,6 +479,57 @@ <h4><a class="toc-backref" href="#id11">num.stream.threads</a><a class="headerli
...
+              <p>Here is an example that adjusts the memory size consumed by RocksDB.</p>
+              <div class="highlight-java"><div class="highlight"><pre><span></span>    <span class="kd">public</span> <span class="kd">static</span> <span class="kd">class</span> <span class="nc">CustomRocksDBConfig</span> <span class="kd">implements</span> <span class="n">RocksDBConfigSetter</span> <span class="o">{</span>
+                    <span class="c1">// This object should be a member variable so it can be closed in RocksDBConfigSetter#close.</span>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:489]: `can be closed` -> `should be closed` (or `must be closed`) ?

File: docs/streams/developer-guide/config-streams.html [hunk 5]
```diff
@@ -474,6 +479,57 @@ <h4><a class="toc-backref" href="#id11">num.stream.threads</a><a class="headerli
...
+                    <span class="kd">private</span> <span class="n">org.rocksdb.Cache</span> <span class="n">cache</span> <span class="o">=</span> <span class="k">new</span> <span class="n">org</span><span class="o">.</span><span class="na">rocksdb</span><span class="o">.</span><span class="na">LRUCache</span><span class="o">(</span><span class="mi">16</span> <span class="o">*</span> <span class="mi">1024L</span> <span class="o">*</span> <span class="mi">1024L</span><span class="o">);</span>
+
+                    <span class="nd">@Override</span>
+                    <span class="kd">public</span> <span class="kt">void</span> <span class="nf">setConfig</span><span class="o">(</span><span class="kd">final</span> <span class="n">String</span> <span class="n">storeName</span><span class="o">,</span> <span class="kd">final</span> <span class="n">Options</span> <span class="n">options</span><span class="o">,</span> <span class="kd">final</span> <span class="n">Map</span><span class="o">&lt;</span><span class="n">String</span><span class="o">,</span> <span class="n">Object</span><span class="o">&gt;</span> <span class="n">configs</span><span class="o">)</span> <span class="o">{</span>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:493]: I cannot really read this part with all the markup... :( I trust you that it's fine :)

File: docs/streams/developer-guide/config-streams.html [hunk 6]
```diff
@@ -684,70 +728,64 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
             </tr>
-            <tr class="row-odd"><td>rocksdb.config.setter</td>
-              <td>Consumer</td>
-              <td>&nbsp;</td>
-            </tr>
             </tbody>
           </table>
         </div>
+        <div class="section" id="parameters-controlled-by-kafka-streams">
+          <h3><a class="toc-backref" href="#id26">Required configuration parameters</a><a class="headerlink" href="#parameters-controlled-by-kafka-streams" title="Permalink to this headline"></a></h3>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: `Required` ? I thought those are the one a user cannot set?

File: docs/streams/developer-guide/config-streams.html [hunk 7]
```diff
@@ -684,70 +728,64 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
+              </tr>
+              </thead>
+              <tbody valign="top">
+              <tr class="row-even"><td>auto.create.topics.enable</td>
+              <td>Consumer</td>
+              <td>false</td>
+              </tr>
+              <tr class="row-even"><td>auto.offset.reset</td>
+              <td>Consumer</td>
+              <td>earliest</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html:764]: Users can overwrite `auto.offset.reset` -- it's just a different default that for the plain `KafkaConsumer` -- same for the following 4 configs.

File: docs/streams/developer-guide/config-streams.html [hunk 8]
```diff
@@ -684,70 +728,64 @@ <h4><a class="toc-backref" href="#id18">Default Values</a><a class="headerlink"
...
+              <col width="31%">
+              </colgroup>
+              <thead valign="bottom">
+              <tr class="row-odd"><th class="head">Parameter Name</th>
+              <th class="head">Corresponding Client</th>
+              <th class="head">Streams Default</th>
+              </tr>
+              </thead>
+              <tbody valign="top">
+              <tr class="row-even"><td>auto.create.topics.enable</td>
```
Comment (mjsax) [inline docs/streams/developer-guide/config-streams.html]: The config name is `allow.auto.create.topics`

File: docs/streams/architecture.html (hunk 1 above)
Comment (JimGalasyn) [reply docs/streams/architecture.html]: I'm not sure either! :) It's what we ended up with in https://github.com/confluentinc/docs/pull/3604.

File: docs/streams/developer-guide/config-streams.html (hunk 2 above)
Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html:272]: Good question, I added it just to keep the style consistent.

File: docs/streams/developer-guide/config-streams.html (hunk 6 above)
Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html]: D'oh! The perils of copy-paste.

**Comment by JimGalasyn** [comment]
Thanks for the review, @mjsax!

File: docs/streams/developer-guide/config-streams.html (hunk 7 above)
Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html:764]: I reworked this a bit, is it accurate?

File: docs/streams/architecture.html (hunk 1 above)
Comment (mjsax) [reply docs/streams/architecture.html]: @JimGalasyn Can we update this to (also in CP docs):
```
Topic partitions are assigned to tasks, and tasks are assigned to all threads over all instances in an best-effort manner try to trade-off load-balancing and stickiness of stateful tasks. For this assignment, Kafka Streams uses ...
```

Comment (JimGalasyn) [reply docs/streams/architecture.html]: Much better, thanks!

**Comment by JimGalasyn** [comment]
Thanks, @mjsax !

**Pull request merged**